### PR TITLE
using jaro-winkler, see if it passes pytest

### PR DESCRIPTION
**Pull request by nrposner**

### CODE REVIEW

**Review by averyschoen — APPROVED** [review]

File: utils/linkage.py [hunk 1]
```diff
@@ -24,86 +26,8 @@ def calculate_string_similarity(string1: str, string2: str) -> float:
...
-    Traceback (most recent call last):
-        ...
-    ValueError: address_line_1 must have whitespace
-    >>> get_street_from_address_line_1("PO Box 1111")
-    Traceback (most recent call last):
-        ...
-    ValueError: address_line_1 is PO Box
-    """
-    pass
+    return float(td.jaro_winkler(string1.lower(), string2.lower()))
```
Comment (averyschoen) [inline utils/linkage.py]: push to think about how to best handle semantic similarity in our usage context...

**Review by averyschoen — CHANGES_REQUESTED** [review]

**Comment by nrposner** [comment]
Two thoughts:
 - in our corpus of names, we should expect non-matches and variants for first names (bob vs robert) more often than for surnames, which tend to be stable for administrative purposes. Jaro-Winkler prioritizes matches in the early characters of two strings. This is not especially useful for us. 
 - we might think of running jar-winkler on substrings, comparing first and last names, for example. However, in cases where the number of substrings don't match up, this is less useful. 
 - We can improve performance easily by reversing the strings: consider that the JW score of 'bob smith' and 'robert smith' is just 0.72, while 'anantarya smith' and 'robert smith' have a score of .69. If we reverse the strings, this goes to .91 vs .69. This also helps us prioritize surnames, which are more stable and often more identifying.
 - another route would be to switch to a substring matching algorithm. These tend to be slower, but this may not be too much of a problem if we plan to run this relatively rarely.

**Comment by nrposner** [comment]
> Two thoughts:
> 
> * in our corpus of names, we should expect non-matches and variants for first names (bob vs robert) more often than for surnames, which tend to be stable for administrative purposes. Jaro-Winkler prioritizes matches in the early characters of two strings. This is not especially useful for us.
> * we might think of running jar-winkler on substrings, comparing first and last names, for example. However, in cases where the number of substrings don't match up, this is less useful.
> * We can improve performance easily by reversing the strings: consider that the JW score of 'bob smith' and 'robert smith' is just 0.72, while 'anantarya smith' and 'robert smith' have a score of .69. If we reverse the strings, this goes to .91 vs .69. This also helps us prioritize surnames, which are more stable and often more identifying.
> * another route would be to switch to a substring matching algorithm. These tend to be slower, but this may not be too much of a problem if we plan to run this relatively rarely.

This would also help for address names: the cities and street names of addresses are more identifying than street numbers.

**Comment by nrposner** [comment]
I may also consider making a variant of jaro-winkler: the current prefix limit is 4 characters, while we may benefit from longer matches at the end. An algorithm that weights matches more strongly based on how close they are to the beginning or end could work.

**Review by averyschoen — APPROVED** [review]

**Pull request merged**